source/features/show-whitespace.tsx

```diff
@@ -15,6 +15,9 @@ function showWhiteSpacesOn(line: Element): void {
 	for (const [nodeIndex, textNode] of textNodesOnThisLine.entries()) {
 		// `textContent` reads must be cached #2737
 		let text = textNode.textContent!;
+		if (text.length > 10_000) { // #5092
+			continue;
+		}
```

Comment:
This limit is very much arbitrary and probably too high.

Should we also check the number of text nodes on the line? Can there be a situation with a huge amount of very short text nodes, or is this impossible?

Reply:
1000 would be enough IMO